src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,157 @@
+
+        given:
+        List<Item> testListItem = [food, gadget, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        !isEligible
+    }
+
+    def "isCartEligibleForDiscount should respond with false if cart price is greater than 50.0 but cart size is less than 5"() {
+
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        !isEligible
+    }
+
+    def "isCartEligibleForDiscount should respond with false if cart price is less than 50.0 but cart size is greater than 5"() {
+
+        given:
+        List<Item> testListItem = [food2, gadget2, clothing2, gadget, clothing, appliance2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        !isEligible
+    }
+
+    @Unroll
+    def "ApplyDiscountToCartItems should respond with discounted cost"() {
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing, food2, gadget2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        orderingService.applyDiscountToCartItems(testCart)
+
+        then:
+        testCart.items.cost << [1.5,1,2,1,0.5,0.5]
+    }
+
+    def "CreateAnOrder should throw exception when canContainFood is true and cart has no food items"() {
+        given:
+        List<Item> testListItem = [gadget, appliance, clothing, gadget2, food]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+        def recipientName = "Harry"
+        def recipientAddress = "Cebu City"
+        def canContainFood = false
+
+        when:
+        def testOrder = orderingService.createAnOrder(testCart, recipientName, recipientAddress, canContainFood)
+
+        then:
+        UnableToCreateOrderException orderException = thrown(UnableToCreateOrderException)
+        orderException.message == "Cart should not contain FOOD items"
+    }
+
+    def "CreateAnOrder should save new order and set order status to Pending"() {
+        given:
+        List<Item> testListItem = [gadget, appliance, clothing, gadget2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+        def recipientName = "Harry"
+        def recipientAddress = "Cebu City"
+        def canContainFood = true
+
+        when:
+        def testOrder = orderingService.createAnOrder(testCart, recipientName, recipientAddress, canContainFood)
+
+        then:
+        testOrder.totalCost == orderingService.calculateTotalCostOfCart(testCart)
```

Comment:
We shouldn't call other service methods here and always try to make things simple. Try to assert it with the actual expected value instead.